elm/model_selection/cross_validation.py

```diff
@@ -0,0 +1,61 @@
+from dask_searchcv.methods import CVCache
+from sklearn.base import BaseEstimator
+import numpy as np
+
+class CVCacheSampler(BaseEstimator, CVCache):
+    def __init__(self, sampler, splits=None, pairwise=None, cache=None):
+        self.sampler = sampler
+        super(CVCacheSampler, self).__init__(splits, pairwise=pairwise,
+                                              cache=cache)
+
+    def _post_splits(self, X, y=None, n=None, is_x=True, is_train=False):
+        if self.splits is None:
+            raise ValueError('Expected .splits to before _post_splits')
+        if y is not None:
+            raise ValueError('y should be None (found {})'.format(type(y)))
+        func = getattr(self.sampler, 'fit_transform', None)
+        if func is None:
+            func = getattr(self.sampler, 'transform', self.sampler)
+        return func(X, y=y, is_x=is_x, is_train=is_train)
+
+    def _extract(self, X, y, n, is_x=True, is_train=True):
+        if self.cache is not None and (n, is_x, is_train) in self.cache:
+            return self.cache[n, is_x, is_train]
+
+        inds = self.splits[n][0] if is_train else self.splits[n][1]
+
+        post_splits = getattr(self, '_post_splits', None)
```

Comment:
@TomAugspurger This is a work in progress still - moving the `post_splits` related CV/sampler logic from https://github.com/dask/dask-searchcv/pull/61 to this new Elm PR that has Elm #221 as a base branch